### PR TITLE
Mailing List Touchup

### PR DESCRIPTION
**Pull request by fpunny**
- Fixed buggy input (We use models now :D only took me till 3:30am to learn/implement...)
 - Replaced axios with our custom query thingy which should be cleaner i hope (axios is evil >:c)
 - Moved email validator into own file because its a very common thingy
 - Made a checkbox component
 - Removed auto navigate from sub/unsub page, now there is a button to go back

### CODE REVIEW

**Review by alerobertson — APPROVED** [review]
I'm gonna change the unsubscribe params but that doesn't affect this so all good.

File: src/pages/CardLayout/Subscribe.vue [hunk 1]
```diff
@@ -90,25 +98,22 @@
...
+          await query(`
+            mutation sub($email: String!, $source: String) {
+              subscribeMailingList(email: $email, source: $source) {
+                message
+              }
+            }
+          `, { email: this.email, source: this.source });
+          this.next();
+        } catch (err) {
+          alert(err);
```
Comment (lyninx) [inline src/pages/CardLayout/Subscribe.vue:115]: `alert` ...yikes

**Review by lyninx — APPROVED** [review]
looks good! 💯

File: src/pages/CardLayout/Subscribe.vue (hunk 1 above)
Comment (alerobertson) [reply src/pages/CardLayout/Subscribe.vue:115]: it made debugging easy lol

Comment (fpunny) [reply src/pages/CardLayout/Subscribe.vue:115]: i wanted console.error, but vue no like me xD (I need to get that modal going)

Comment (lyninx) [reply src/pages/CardLayout/Subscribe.vue:115]: i think we should de prioritize the modal for now though. hacker dashboard is most important right now!

**Comment by lyninx** [comment]
can we ship this?

**Pull request merged**